core/app/models/spree/promotion_code/code_builder.rb

```diff
@@ -26,7 +26,7 @@ def build_promotion_codes
   private
 
   def codes
-    if num_codes > 1
+    if (num_codes + promotion.codes.size) > 1
```

Comment:
It's a bit unclear to me why this change is required. A more detailed commit message would be welcome.